mender-update/deployments.cpp

```diff
@@ -314,6 +315,93 @@ error::Error PushStatus(
 		});
 }
 
+static const string logs_uri_suffix = "/log";
+
+error::Error PushLogs(
+	const string &deployment_id,
+	LogMessages messages,
+	const string &server_url,
+	http::Client &client,
+	events::EventLoop &loop,
+	LogsAPIResponseHandler api_handler) {
+	if (messages.size() == 0) {
+		// nothing to do
+		return error::NoError;
```

Comment:
I would argue that returning no error, but then not calling `api_handler`, goes against our typical async behavior, and could be surprising. I have three suggestions:

1. Return a "nothing-to-do" kind of error.
2. Send a log anyway, consisting of one message, "No messages".
3. Call `api_handler` right away with success. In this case it can probably be called directly, instead of using `Post` since there is no waiting involved.

I don't have a strong preference, but when deciding, keep in mind that this is extremely unlikely to happen in practice, because even just the client itself will produce messages before an update even gets going.

Reply:
> not calling `api_handler`, goes against our typical async behavior, and could be surprising

100% agree, I've been trying to keep an eye on this myself. But briefly forgot about it when I did this optimization/simplification. I like the 3rd option the best, it makes most sense to me.